src/js/crud.js

```diff
@@ -0,0 +1,168 @@
+// Check if authorized
+if (sessionStorage.getItem("auth") !== "granted") {
+  alert("Unauthorized access!");
+  window.location.href = "login.html";
+}
+
+import { fetchFromMultipleBases, getBaseUrls } from './fetch.js';
+
+// Helper for base URL
+async function getBaseUrl(endpoint) {
+  const bases = await getBaseUrls();
+  return `${bases[0]}/${endpoint}`;
+}
+
+// Predefined tags
+const predefinedTags = [
+  "bread", "artisan", "savory", "vegan", "special!", 
+  "pastry", "buttery", "breakfast", "sweet", "dessert",
+  "chocolate", "fruit", "snack", "cake", "citrus"
+];
+
+// Helpers
+function highlightError(inputEl) {
+  inputEl.style.borderColor = "red";
+  inputEl.focus();
+}
+function clearErrors(inputs) {
+  inputs.forEach(i => i.style.borderColor = "#ccc");
+}
+
+// Populate tags dropdown
+function populateTagsDropdown() {
+  const tagsSelect = document.getElementById("tags");
+  predefinedTags.forEach(tag => {
+    const option = document.createElement("option");
+    option.value = tag;
+    option.textContent = tag;
+    tagsSelect.appendChild(option);
+  });
+}
+
+// Fetch and render products
+async function fetchProducts() {
+  const bases = await getBaseUrls();
+  const products = await fetchFromMultipleBases('products', bases);
+
+  const list = document.querySelector('.product-list');
+  list.innerHTML = ""; // clear old list
+
+  products.forEach(p => {
```

Comment:
Hier heb je geen optional chaining toegepast. En dus ook geen check of er uit deze variable wel een array komt